### ISSUE TITLE
How to trim video without downloading it entirely?

### ISSUE THREAD

**Issue by Mennaruuk** [yt-dlp/yt-dlp#2220]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

I noticed that when I want to use start and end times to trim a video, the video has to be downloaded first in its entirety before the trimming occurs.

Is there a way to have yt-dlp download the desired portion only?

Thank you.

**Comment by pukkandan** [comment]
https://github.com/yt-dlp/yt-dlp/issues/686#issuecomment-904191812

tl;dr

    --downloader ffmpeg --downloader-args "ffmpeg_i:-ss start_time -to end_time"

**Comment by Mennaruuk** [comment]
THANK YOU! It worked marvelously :)

Here's the command I ran: `yt-dlp https://www.youtube.com/watch?v=xMOHrl_is4Y --downloader ffmpeg --downloader-args "ffmpeg_i:-ss 997 -to 2512"`

**Comment by yamete0** [comment]
and this fucking shit is fucking broken. fucking amateur program. cant download a basic fucking timeframe without fucking up the quality or time

**Comment by xfeeefeee** [comment]
since this is one of the first google search results that comes up when you search for it, the modern way to do this with yt-dlp is to use the `--download-sections` option

```
--download-sections REGEX       Download only chapters whose title matches
                                the given regular expression. Time ranges
                                prefixed by a "*" can also be used in place
                                of chapters to download the specified range.
                                Needs ffmpeg. This option can be used
                                multiple times to download multiple
                                sections, e.g. --download-sections
                                "*10:15-inf" --download-sections "intro"
```

**Comment by srghma** [comment]
`yt-dlp https://www.youtube.com/watch\?v\=g1Sdngywob4 --download-sections "*17:16-19:31"` downloaded my video without audio

but 

`yt-dlp https://www.youtube.com/watch\?v\=g1Sdngywob4 --downloader ffmpeg --downloader-args "ffmpeg_i:-ss 1036 -to 1171"` - with audio - no problems

**Comment by bashonly** [comment]
@srghma try adding `-S proto:https`

**Comment by jtheletter** [comment]
Anyone yet learned how to do this?

**Comment by Offroaders123** [comment]
> Anyone yet learned how to do this?

I use a similar command to that of the response that Mennaruuk wrote before closing the issue. It's in combination with ffmpeg, which is what allows it to work for us.

> Here's the command I ran: `yt-dlp https://www.youtube.com/watch?v=xMOHrl_is4Y --downloader ffmpeg --downloader-args "ffmpeg_i:-ss 997 -to 2512"`

You can also instead format the time parameters with `hh:mm:ss`, that is easier for me to conceptualize personally. So say if it started at the 10:07 mark, you would put `-ss 00:10:07` as the first parameter.

**Comment by bashonly** [comment]
Or just use yt-dlp's `--download-sections` option as has been suggested multiple times in this issue

**Comment by doncezart** [comment]
> @ivo-ns use `--download-sections "*31:26-36:16" -S proto:https`

A few months late but encountering this issue as well - your command only outputs a webm audio file, no video
`yt-dlp url --download-sections "*2:45:00-2:46:14" -S proto:https`
Same with the ffmpeg approach
`yt-dlp url --downloader ffmpeg --downloader-args "ffmpeg_i:-ss 2:45:00 -to 2:46:14"`
Am I doing something wrong?

**Comment by aymen157** [comment]
Seems like `yt-dlp.exe --download-sections "*17:51-18:13" YTB_VIDEO_URL` misses video (ie its audio only), the solution for whatever reason was as @bashonly suggested `-S proto:https`

**Comment by SoroushAU** [comment]
> since this is one of the first google search results that comes up when you search for it, the modern way to do this with yt-dlp is to use the `--download-sections` option

An option to use seconds instead of time expressions would be nice since it's just easier to work with seconds when implementing stuff programatically. Guess I'll just continue using `--downloader-args "ffmpeg_i:-ss 997 -to 2512"`.

**Comment by Offroaders123** [comment]
I think you can still pass in `hh:mm:ss` timestamps for that, so it should still allow you to use seconds with that.

**Comment by avedenyev** [comment]
You can use -f <format> to point to the video/audio format you want to download
yt-dlp -f 140 "https://www.youtube.com/watch?v=xxx" --downloader ffmpeg --downloader-args "ffmpeg_i:-ss 0 -to 600"
It's worth to mention that time is in seconds. This download is slow, it takes the same amount of time as it plays.

**Comment by IArnaut2** [comment]
It doesn't trim video from specific time. I put `-ss 8 -to 28` or `-ss 00:00:08 -to 00:00:28", but the result does not start from 8 seconds of the original video, but from the start of the original video.
Result:
```
Input #0, matroska,webm, from 'https://rr1---sn-vgpvo-cxbl.googlevideo.com/videoplayback?expire=1756918494&ei=fh64aNT1IdehzPsPwrW_wQs&ip=93.186.72.11&id=o-AI1tRhWXjI5Oo59-o-Snkz4xJd5PsDlP9jIClp0elyF7&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1756896894%2C&mh=80&mm=31%2C29&mn=sn-vgpvo-cxbl%2Csn-nv47lnsr&ms=au%2Crdu&mv=m&mvi=1&pl=24&rms=au%2Cau&initcwndbps=2342500&bui=AY1jyLNfKhJFfH2BBsTBZdYO-AExB-TaekQ33LCLzSo2xDfLAYObwflXF81o9301RJA1gbJwpyq5vbhu&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=iXRa36nVlZ0ir36kgyq0fowQ&rqh=1&gir=yes&clen=616239&dur=37.101&lmt=1720901801145540&mt=1756896372&fvip=2&keepalive=yes&lmw=1&fexp=51557447%2C51565116%2C51565681%2C51580970&c=TVHTML5&sefc=1&txp=6308224&n=nYHpg5_eYBBqyA&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgLg-b6aNonvEjZQ4kebNX2qFwaUhz3x2KndLCSoJECGQCIQCRuHeZiuiDllQ7sZrt1th9AWRNmwb2qod9uowOr1qUgg%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgWaNcP55BWHxrDdwv_HL4MJ3Q03tUkz6SDM09qSA7o5ICIFtqiSTed7y4CpcBHX4aRXci0WhxHkpIlTGPhHQ2gNqe':
  Metadata:
    encoder         : google/video-file
  Duration: 00:00:37.10, start: -0.007000, bitrate: 132 kb/s
  Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
Output #0, webm, to 'file:Half-life mobile ads be like 2 [YBB8kqAJSmQ].webm.part':
  Metadata:
    encoder         : Lavf61.9.100
  Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
Press [q] to stop, [?] for help
[out#0/webm @ 000001c029e48180] video:0KiB audio:439KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 2.320837%
size=     449KiB time=00:00:20.00 bitrate= 184.0kbits/s speed= 118x
```

**Comment by IArnaut2** [comment]
> I think you can still pass in `hh:mm:ss` timestamps for that, so it should still allow you to use seconds with that.

Problem is, I type `08` for the start time, but the result does NOT start from 8 seconds of the original video.

**Comment by q3m** [comment]
Encountering the same issue of start times not matching. I was able to use `--force-keyframes-at-cuts` as a workaround, but it is considerably slower. Also had to add `--downloader-args "ffmpeg_i1:-reconnect 1" --downloader-args "ffmpeg_i2:-reconnect 1"` (taken from [this reddit post](https://www.reddit.com/r/youtubedl/comments/1ciw2a2/my_solution_for_io_error_error_number_10054/)) for longer downloads to keep it from disconnecting.

**Comment by Road2Redemption** [comment]
> It doesn't trim video from specific time. I put `-ss 8 -to 28` or `-ss 00:00:08 -to 00:00:28", but the result does not start from 8 seconds of the original video, but from the start of the original video. Result:
> 
> ```
> Input #0, matroska,webm, from 'https://rr1---sn-vgpvo-cxbl.googlevideo.com/videoplayback?expire=1756918494&ei=fh64aNT1IdehzPsPwrW_wQs&ip=93.186.72.11&id=o-AI1tRhWXjI5Oo59-o-Snkz4xJd5PsDlP9jIClp0elyF7&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1756896894%2C&mh=80&mm=31%2C29&mn=sn-vgpvo-cxbl%2Csn-nv47lnsr&ms=au%2Crdu&mv=m&mvi=1&pl=24&rms=au%2Cau&initcwndbps=2342500&bui=AY1jyLNfKhJFfH2BBsTBZdYO-AExB-TaekQ33LCLzSo2xDfLAYObwflXF81o9301RJA1gbJwpyq5vbhu&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=iXRa36nVlZ0ir36kgyq0fowQ&rqh=1&gir=yes&clen=616239&dur=37.101&lmt=1720901801145540&mt=1756896372&fvip=2&keepalive=yes&lmw=1&fexp=51557447%2C51565116%2C51565681%2C51580970&c=TVHTML5&sefc=1&txp=6308224&n=nYHpg5_eYBBqyA&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgLg-b6aNonvEjZQ4kebNX2qFwaUhz3x2KndLCSoJECGQCIQCRuHeZiuiDllQ7sZrt1th9AWRNmwb2qod9uowOr1qUgg%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgWaNcP55BWHxrDdwv_HL4MJ3Q03tUkz6SDM09qSA7o5ICIFtqiSTed7y4CpcBHX4aRXci0WhxHkpIlTGPhHQ2gNqe':
>   Metadata:
>     encoder         : google/video-file
>   Duration: 00:00:37.10, start: -0.007000, bitrate: 132 kb/s
>   Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
> Stream mapping:
>   Stream #0:0 -> #0:0 (copy)
> Output #0, webm, to 'file:Half-life mobile ads be like 2 [YBB8kqAJSmQ].webm.part':
>   Metadata:
>     encoder         : Lavf61.9.100
>   Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
> Press [q] to stop, [?] for help
> [out#0/webm @ 000001c029e48180] video:0KiB audio:439KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 2.320837%
> size=     449KiB time=00:00:20.00 bitrate= 184.0kbits/s speed= 118x
> ```

Hi, I have met the same problem.  GPT tells me that HLS downloads many 'subsections', for example, 0-10 is a subsection, but you want to download from 8-28, so it will automatically download 0-10, 10-20, and 20-28.

I am not sure whether this is correct or not, feel free to correct me. 

And do you have any solution?

**Comment by Matojeje** [comment]
I couldn't find a way to do this without re-encoding the video, unfortunately :c

**Comment by IArnaut2** [comment]
> > It doesn't trim video from specific time. I put `-ss 8 -to 28` or `-ss 00:00:08 -to 00:00:28", but the result does not start from 8 seconds of the original video, but from the start of the original video. Result:
> > ```
> > Input #0, matroska,webm, from 'https://rr1---sn-vgpvo-cxbl.googlevideo.com/videoplayback?expire=1756918494&ei=fh64aNT1IdehzPsPwrW_wQs&ip=93.186.72.11&id=o-AI1tRhWXjI5Oo59-o-Snkz4xJd5PsDlP9jIClp0elyF7&itag=251&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1756896894%2C&mh=80&mm=31%2C29&mn=sn-vgpvo-cxbl%2Csn-nv47lnsr&ms=au%2Crdu&mv=m&mvi=1&pl=24&rms=au%2Cau&initcwndbps=2342500&bui=AY1jyLNfKhJFfH2BBsTBZdYO-AExB-TaekQ33LCLzSo2xDfLAYObwflXF81o9301RJA1gbJwpyq5vbhu&vprv=1&svpuc=1&mime=audio%2Fwebm&ns=iXRa36nVlZ0ir36kgyq0fowQ&rqh=1&gir=yes&clen=616239&dur=37.101&lmt=1720901801145540&mt=1756896372&fvip=2&keepalive=yes&lmw=1&fexp=51557447%2C51565116%2C51565681%2C51580970&c=TVHTML5&sefc=1&txp=6308224&n=nYHpg5_eYBBqyA&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cbui%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgLg-b6aNonvEjZQ4kebNX2qFwaUhz3x2KndLCSoJECGQCIQCRuHeZiuiDllQ7sZrt1th9AWRNmwb2qod9uowOr1qUgg%3D%3D&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=APaTxxMwRAIgWaNcP55BWHxrDdwv_HL4MJ3Q03tUkz6SDM09qSA7o5ICIFtqiSTed7y4CpcBHX4aRXci0WhxHkpIlTGPhHQ2gNqe':
> >   Metadata:
> >     encoder         : google/video-file
> >   Duration: 00:00:37.10, start: -0.007000, bitrate: 132 kb/s
> >   Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
> > Stream mapping:
> >   Stream #0:0 -> #0:0 (copy)
> > Output #0, webm, to 'file:Half-life mobile ads be like 2 [YBB8kqAJSmQ].webm.part':
> >   Metadata:
> >     encoder         : Lavf61.9.100
> >   Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
> > Press [q] to stop, [?] for help
> > [out#0/webm @ 000001c029e48180] video:0KiB audio:439KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 2.320837%
> > size=     449KiB time=00:00:20.00 bitrate= 184.0kbits/s speed= 118x
> > ```
> 
> Hi, I have met the same problem. GPT tells me that HLS downloads many 'subsections', for example, 0-10 is a subsection, but you want to download from 8-28, so it will automatically download 0-10, 10-20, and 20-28.
> 
> I am not sure whether this is correct or not, feel free to correct me.
> 
> And do you have any solution?

No, not at all.

**Comment by bashonly** [comment]
The starting point of any video section will always be the nearest preceding keyframe unless you re-encode (i.e. use `--force-keyframes-with-cuts`)
